account_ecotax/tests/test_ecotax.py

```diff
@@ -0,0 +1,544 @@
+                "ecotax_line_product_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "classification_id": ecotax_classification.id,
+                        },
+                    )
+                ],
+                # For the sake of simplicity, every product will have a price
+                # and weight of 100
+                "list_price": 100.00,
+                "weight": 100.00,
+                "attribute_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "attribute_id": size_attr.id,
+                            "value_ids": [(6, 0, size_attr.value_ids.ids)],
+                        },
+                    )
+                ],
+            }
+        )
+        return tmpl.product_variant_ids
+
+    @staticmethod
+    def _set_invoice_lines_random_quantities(invoice) -> list:
+        """For each invoice line, sets a random qty between 1 and 10
+
+        Returns the list of new quantities as a list
+        """
+        new_qtys = []
+        with Form(invoice) as invoice_form:
+            for index in range(len(invoice.invoice_line_ids)):
+                with invoice_form.invoice_line_ids.edit(index) as line_form:
+                    new_qty = choice([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+                    line_form.quantity = new_qty
+                    new_qtys.insert(index, new_qty)
+                line_form.save()
+        invoice_form.save()
+        return new_qtys
+
+    def _run_checks(self, inv, inv_expected_amounts, inv_lines_expected_amounts):
+        self.assertEqual(inv.amount_ecotax, inv_expected_amounts["amount_ecotax"])
+        self.assertEqual(inv.amount_total, inv_expected_amounts["amount_total"])
+        self.assertEqual(len(inv.invoice_line_ids), len(inv_lines_expected_amounts))
+        for inv_line, inv_line_expected_amounts in zip(
+            inv.invoice_line_ids, inv_lines_expected_amounts
+        ):
+            self.assertEqual(
+                inv_line.ecotax_amount_unit,
+                inv_line_expected_amounts["ecotax_amount_unit"],
+            )
+            self.assertEqual(
+                inv_line.subtotal_ecotax, inv_line_expected_amounts["subtotal_ecotax"]
+            )
+
+    def test_01_default_fixed_ecotax(self):
+        """Test default fixed ecotax
+
+        Ecotax classification data for this test:
+            - fixed type
+            - default amount: 5.0
+        Product data for this test:
+            - list price: 100
+            - fixed ecotax
+            - no manual amount
+
+        Expected results (with 1 line and qty = 1):
+            - invoice ecotax amount: 5.0
+            - invoice total amount: 100.0
+            - line ecotax unit amount: 5.0
+            - line ecotax total amount: 5.0
+        """
```

Comment:
why invoice total amount `100` and not `120`?

Reply:
It seems to me you're right. @mourad-ehm will have a look tomorrow

Reply:
Ecotax is included in list_price (it's part of it)
So yes, invoice total amount is 100.